Extension/src/LanguageServer/client.ts

```diff
@@ -987,6 +1076,17 @@ export class DefaultClient implements Client {
                     } else {
                         this.configuration.CompilerDefaults = compilerDefaults;
                     }
+                    // update all client configurations
+                    this.configuration.setupConfigurations();
```

Comment:
I had commented here in the context of one of Sean's comments, but it looks like he resolved it.

This may be displaying the prompt too earlier.  If the user is using a `compile_commands.json` file, we will not know about those compilers used until after the configuration has been processed. 

We do process configurations in `setupConfigurations`, immediately before this.  However,   `ChangeCppPropertiesNotification` is a notification, so the TypeScript code does not wait for it to complete before proceeding.  A simple solution might be to change that message from a notification to request and await it, so that the TypeScript code does not proceed until the configurations have been processed and any discovered compilers in `compile_commands.json` reported to the TypeScript side.

Reply:
We can take a look at this in a future PR fix